src/ucs/sys/sock.c

```diff
@@ -90,13 +90,14 @@ unsigned ucs_netif_bond_ad_num_ports(const char *bond_name)
 
     status = ucs_read_file_number(&ad_num_ports, 1,
                                   UCS_NETIF_BOND_AD_NUM_PORTS_FMT, bond_name);
-    if (status != UCS_OK) {
-        ucs_diag("failed to read " UCS_NETIF_BOND_AD_NUM_PORTS_FMT ": %m, "
+    if ((status != UCS_OK) || ((ad_num_ports <= 0) ||
+        (ad_num_ports > UINT_MAX))) {
```

Comment:
No need for additional `()` around `ad_num_ports` check
Or is this for readability?

Reply:
@dmitrygx it was a copy/paste issue, fixed. 10x!

Reply:
> @dmitrygx it was a copy/paste issue, fixed. 10x!

;)